File: frontend/src/types/recurrence.ts
export interface RecurrenceRule {
  id: number
  issue_id: number
  frequency: 'daily' | 'weekly' | 'monthly' | 'cron'
  interval: number
  cron_expr?: string
  next_run: string
  end_date?: string
  is_active: boolean
  created_at: string
}

export interface RecurrenceCreate {
  frequency: string
  interval?: number
  cron_expr?: string
  next_run: string
  end_date?: string
}

export interface RecurrenceUpdate {
  frequency?: string
  interval?: number
  cron_expr?: string
  next_run?: string
  end_date?: string
  is_active?: boolean
}

File: backend/internal/dto/request/recurrence.go
package request

// RecurrenceCreateRequest creates a recurrence rule.
type RecurrenceCreateRequest struct {
	Frequency string  `json:"frequency" binding:"required"` // daily | weekly | monthly | cron
	Interval  int     `json:"interval"`
	CronExpr  *string `json:"cron_expr"`
	NextRun   string  `json:"next_run"` // RFC3339
	EndDate   *string `json:"end_date"`
}

// RecurrenceUpdateRequest updates a recurrence rule.
type RecurrenceUpdateRequest struct {
	Frequency *string `json:"frequency"`
	Interval  *int    `json:"interval"`
	CronExpr  *string `json:"cron_expr"`
	NextRun   *string `json:"next_run"`
	EndDate   *string `json:"end_date"`
	IsActive  *bool   `json:"is_active"`
}
